### PR TITLE
Shorten mlab tests.

### PR DESCRIPTION
**Pull request by anntzer**
When there's approximately 3x more lines for defining inputs and outputs
than for the actual testing, the purpose of the tests becomes difficult
to see.

Not using `parametrize` because it's a bit tricky to define the
inputs/outputs in the decorator, and adding more indirection would
defeat the purpose of the PR.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/tests/test_mlab.py [hunk 1]
```diff
@@ -80,33 +80,14 @@ def test_stride_windows_n32_noverlap0_unflatten(self, axis):
...
-
-        assert_allclose(targ, res, atol=1e-06)
-
-    def test_window_hanning_ones(self):
-        targ = np.hanning(len(self.sig_ones))
-        res = mlab.window_hanning(self.sig_ones)
-
-        assert_allclose(targ, res, atol=1e-06)
+def test_window():
+    n = 1000
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_mlab.py:84]: Why are we dropping the call to `np.random.seed(0)`?

Comment (anntzer) [reply lib/matplotlib/tests/test_mlab.py:84]: I accidentally dropped it, but again if you looked at the content of the tests it's clear that this should *always* pass regardless of the seed.

Comment (dopplershift) [reply lib/matplotlib/tests/test_mlab.py:84]: True, but in the unlikely even that it ever should fail, it's completely unreproducible with the seed unset.

File: lib/matplotlib/tests/test_mlab.py [hunk 2]
```diff
@@ -119,482 +101,148 @@ def setup(self):
...
-                self.sig_base + self.sig_slope,
-                self.sig_base + self.sig_off + self.sig_slope]
-        arrt = [self.sig_base,
-                self.sig_base,
-                self.sig_base + self.sig_slope_mean,
-                self.sig_base + self.sig_slope_mean]
-        input = np.vstack(arri).T
-        targ = np.vstack(arrt).T
+        target = self.sig_base + self.sig_slope_mean
         res = mlab.detrend_mean(input, axis=0)
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_mlab.py:154]: Duplicated with the assert below?

File: lib/matplotlib/tests/test_mlab.py [hunk 3]
```diff
@@ -119,482 +101,148 @@ def setup(self):
...
+        ]:
+            with pytest.raises(ValueError):
+                mlab.detrend_mean(signal, **kwargs)
+
+    def test_detrend_linear_0d(self):
+        assert mlab.detrend_linear(0.) == 0.
+        assert mlab.detrend_linear(5.5) == 0.
+        assert mlab.detrend(5.5, key="linear") == 0.
+        assert mlab.detrend(5.5, key=mlab.detrend_linear) == 0.
+        for sig in [
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_mlab.py:213]: Isn't this part 1D?

File: lib/matplotlib/tests/test_mlab.py [hunk 4]
```diff
@@ -119,482 +101,148 @@ def setup(self):
...
+                self.sig_slope + self.sig_off,
+        ]:
+            self.allclose(mlab.detrend_linear(sig), self.sig_zeros)
 
-    def test_detrend_str_linear_1d_slope_off(self):
+    def test_detrend_str_linear_1d(self):
         input = self.sig_slope + self.sig_off
-        targ = self.sig_zeros
+        target = self.sig_zeros
         res = mlab.detrend(input, key='linear')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_mlab.py:223]: Redundant with the below assert?

Comment (anntzer) [reply lib/matplotlib/tests/test_mlab.py:223]: yes

File: lib/matplotlib/tests/test_mlab.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_mlab.py:154]: yes

File: lib/matplotlib/tests/test_mlab.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_mlab.py:213]: yes

File: lib/matplotlib/tests/test_mlab.py [hunk 5]
```diff
@@ -119,482 +101,148 @@ def setup(self):
...
+        for sig, tgt in [  # 1D.
+                ("zeros", "zeros"),
+                ("base", "base"),
+                ("base off", "base"),
+                ("base slope", "base slope_mean"),
+                ("base slope off", "base slope_mean"),
+        ]:
+            sig = sum(getattr(self, f"sig_{name}") for name in sig.split())
+            tgt = sum(getattr(self, f"sig_{name}") for name in tgt.split())
+            self.allclose(mlab.detrend_mean(sig), tgt)
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/tests/test_mlab.py:149]: yes

**Review by QuLogic — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Pull request merged**